packages/flutter/lib/src/painting/image_cache.dart

```diff
@@ -183,8 +183,9 @@ class ImageCache {
       _checkCacheSize();
     }
     if (maximumSize > 0 && maximumSizeBytes > 0) {
-      _pendingImages[key] = _PendingImage(result, listener);
-      result.addListener(listener);
+      final ImageStreamListener streamListener = ImageStreamListener(listener);
+      _pendingImages[key] = _PendingImage(result, streamListener);
+      result.addListener(streamListener);
```

Comment:
no matching removal?

Reply:
The removal is done inside `_PendingImage`.